packages/aws-cdk-lib/aws-ecs-patterns/lib/fargate/application-load-balanced-fargate-service.ts

```diff
@@ -70,6 +87,12 @@ export class ApplicationLoadBalancedFargateService extends ApplicationLoadBalanc
     } else if (props.taskDefinition) {
       this.taskDefinition = props.taskDefinition;
     } else if (props.taskImageOptions) {
+      this.validateHealthyPercentage('containerCpu', props.containerCpu);
+      this.validateHealthyPercentage('containerMemoryLimitMiB', props.containerMemoryLimitMiB);
+
+      this.validateNotExceeding('containerCpu', props.cpu ?? 256, props.containerCpu);
+      this.validateNotExceeding('containerMemoryLimitMiB', props.memoryLimitMiB ?? 512, props.containerMemoryLimitMiB);
```

Comment:
Why do you specify 256 for cpu and 512 for memory?

It is true that the minimum values for cpu and memory are those values, but it is a bit confusing to pass those values when props.cpu and props.memory are not specified. If you want to validate the minimum value, it would be good to implement it more directly.

Reply:
It specified default values ​​for props.cpu and props.memoryLimitMiB...

https://github.com/aws/aws-cdk/blob/3670d83f40d22b72d883515b14c819ceb69f81df/packages/aws-cdk-lib/aws-ecs-patterns/lib/base/fargate-service-base.ts#L34

https://github.com/aws/aws-cdk/blob/3670d83f40d22b72d883515b14c819ceb69f81df/packages/aws-cdk-lib/aws-ecs-patterns/lib/base/fargate-service-base.ts#L60


In the modified code, 'JsDoc' indicates that it is the default value, as shown below, but I feel that it is difficult to understand.

What is the best way to implement it?

```ts
  /**
   * Throws an error if containerCpu is greater than cpu or is a not positive integer.
   * @param containerCpu The minimum number of CPU units to reserve for the container.
   * @param cpu The number of cpu units used by the task. (default: 256)
   */
  private validateContainerCpu(containerCpu?: number, cpu: number = 256) {
      // ...
 }

  /**
   * Throws an error if containerMemoryLimitMiB is greater than memoryLimitMiB or is a not positive integer.
   * @param containerMemoryLimitMiB The amount (in MiB) of memory to present to the container.
   * @param memoryLimitMiB The amount (in MiB) of memory used by the task. (default: 512)
   */
  private validateContainerMemoryLimitMiB(containerMemoryLimitMiB?: number, memoryLimitMiB: number = 512) {
    // ...
}
```

Reply:
Thanks for clarifying. I commented at another thread, please take a look at it.